### DISCUSSION TITLE
Question on Book 1 Chapter 4.2 `ray_color` function

### DISCUSSION THREAD

**Discussion by LollipopFt** [RayTracing/raytracing.github.io#1056] [Q&A] [answered]
in `ray_color`:
```c
return (1.0-t)*color(1.0, 1.0, 1.0) + t*color(0.5, 0.7, 1.0);
```
i don't understand why it has to be `color(0.5, 0.7, 1.0)` at the end. what does `ray_color` function do exactly?

**Comment by LollipopFt** [comment]
just to add on, i output the image and received a different colour from the png.
![image](https://user-images.githubusercontent.com/62802897/185664641-a2732f21-2345-48bd-941b-ab6c5ed31aee.png)
left: SDL2 renderer (i decided to use a program to render the image instead of ppm), right: png
is this normal?

**Comment by trevordblack** [comment, marked as answer]
Unfortunately, yes, the behavior above is _expected_. It's not normal. But it is expected.

The ppm file format and whatever sdl2 is using have a slightly different color gamma function.

We don't cover this in the book, I'd like to, but I'm not sure if it makes sense.

The book is written with ppm in mind, as we know the color curve really well, but anything rendered out using a different library will just be different.

**Reply by LollipopFt** [reply]
sure, so i will expect my images to turn out different. however, i still don't understand why the book uses the specific values `0.5`, `0.7` and `1.0`. why not `0.0`, `0.5` and `1.0` or some other alternative?

**Reply by hollasch** [reply]
The specific colors chosen are arbitrary, to yield a realistic sky gradient. Feel free to use whatever gradient you like.

**Reply by LollipopFt** [reply]
thanks for explaining!
